src/hotspot/share/cds/cdsConstants.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2014, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+
+#include <cstddef>
+#include "cds.h"
+#include "cds/cdsConstants.hpp"
+#include "cds/dynamicArchive.hpp"
+#include "cds/filemap.hpp"
+
+CDSConst CDSConstants::offsets[] = {
+  { "CDSFileMapHeaderBase::_magic",           (int)offset_of(CDSFileMapHeaderBase, _magic)           },
+  { "CDSFileMapHeaderBase::_crc",             (int)offset_of(CDSFileMapHeaderBase, _crc)             },
+  { "CDSFileMapHeaderBase::_version",         (int)offset_of(CDSFileMapHeaderBase, _version)         },
+  { "CDSFileMapHeaderBase::_space[0]",        (int)offset_of(CDSFileMapHeaderBase, _space)           },
+  { "FileMapHeader::_jvm_ident",              (int)offset_of(FileMapHeader, _jvm_ident)              },
+  { "FileMapHeader::_base_archive_name_size", (int)offset_of(FileMapHeader, _base_archive_name_size) },
+  { "CDSFileMapRegion::_crc",                 (int)offset_of(CDSFileMapRegion, _crc)                 },
+  { "CDSFileMapRegion::_used",                (int)offset_of(CDSFileMapRegion, _used)                },
+  { "DynamicArchiveHeader::_base_region_crc", (int)offset_of(DynamicArchiveHeader, _base_region_crc) }
+};
+
+CDSConst CDSConstants::constants[] = {
+  { "static_magic",                 (int)CDS_ARCHIVE_MAGIC         },
+  { "dynamic_magic",                (int)CDS_DYNAMIC_ARCHIVE_MAGIC },
+  { "int_size",                     sizeof(int)                    },
+  { "CDSFileMapRegion_size",        sizeof(CDSFileMapRegion)       },
```

Comment:
Should be `(int)sizeof(int)` because the type of `sizeof(...)` is implementation-defined. See https://en.cppreference.com/w/cpp/types/size_t

Reply:
Now that all almost all (except for 'magic') with size_t, we should use name vs size_t in the struct and get functions, cast when return from vm in whiteBox.cpp.